profiles/commands.py

```diff
@@ -70,18 +72,34 @@ def _update_affiliations_from_orcid_record(profile: Profile, orcid_record: dict)
 
 
 def _update_email_addresses_from_orcid_record(profile: Profile, orcid_record: dict) -> None:
-    orcid_emails = extract_email_addresses(orcid_record)
+    orcid_email_dicts = extract_email_addresses(orcid_record)
+    profiles = SQLAlchemyProfiles(db)
+
+    for index, email_dict in enumerate(orcid_email_dicts):
+        email = email_dict.get('email')
+        try:
+            email_profile = profiles.get_by_email_address(email)
+        except ProfileNotFound:
+            continue
+
+        if email_profile.id != profile.id:
+            message = ('Profile %s is trying to add email address %s but this '
+                       'email is associated with profile %s which violates '
+                       'unique constraint for email addresses' %
+                       (profile.id, email, email_profile.id))
+            LOGGER.error(message)
```

Comment:
Just being aware of the fact that this error will be logged every time an update for `profile.id` is delivered? But since we don't fail the request, it's not very frequent.

Reply:
It's only logged when the condition is True on Line 85.